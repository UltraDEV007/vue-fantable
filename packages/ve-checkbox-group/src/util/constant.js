// prefix
export const PREFIX_CLS = 've-checkbox-group-'

// emit events
export const EMIT_EVENTS = {
  // on checked change
  ON_CHECKED_CHANGE: 'on-checked-change',
}

// comps name
export const COMPS_NAME = {
  VE_CHECKBOX: 'VeCheckbox',
  VE_CHECKBOX_GROUP: 'VeCheckboxGroup',
}
