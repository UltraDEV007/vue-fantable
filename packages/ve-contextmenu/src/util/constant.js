// prefix
export const PREFIX_CLS = 've-contextmenu-'

// comps name
export const COMPS_NAME = {
  VE_CONTEXTMENU: 'VeContextmenu',
}

// emit events
export const EMIT_EVENTS = {
  ON_NODE_CLICK: 'on-node-click',
}

// init data
export const INIT_DATA = {
  // PARENT_ID: 0,
  PARENT_DEEP: -1,
}

// contextmenu types
export const CONTEXTMENU_NODE_TYPES = {
  // Separator
  SEPARATOR: 'SEPARATOR',
}

// instance methods
export const INSTANCE_METHODS = {
  HIDE_CONTEXTMENU: 'hideContextmenu',
}
