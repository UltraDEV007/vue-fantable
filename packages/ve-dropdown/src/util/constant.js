// prefix
export const PREFIX_CLS = 've-dropdown-'

// emit events
export const EMIT_EVENTS = {
  FILTER_RESET: 'on-filter-reset',
  FILTER_CONFIRM: 'on-filter-confirm',
  VISIBLE_CHANGE: 'on-dropdown-visible-change',
  ITEM_SELECT_CHANGE: 'on-item-select-change',
}

// comps name
export const COMPS_NAME = {
  VE_DROPDOWN: 'VeDropdown',
}
