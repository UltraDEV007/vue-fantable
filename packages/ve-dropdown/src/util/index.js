/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return 've-dropdown-' + cls
}
