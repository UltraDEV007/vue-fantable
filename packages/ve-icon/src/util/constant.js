// prefix
export const PREFIX_CLS = 've-icon-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_ICON: 'VeIcon',
}
