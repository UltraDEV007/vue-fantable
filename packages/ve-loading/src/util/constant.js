// prefix
export const PREFIX_CLS = 've-loading-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_LOADING: 'VeLoading',
  VE_LOADING_PLANE: 'VeLoadingPlane',
  /* VE_LOADING_CHASE: "VeLoadingChase", */
  VE_LOADING_BOUNCE: 'VeLoadingBounce',
  VE_LOADING_WAVE: 'VeLoadingWave',
  VE_LOADING_PULSE: 'VeLoadingPulse',
  VE_LOADING_FLOW: 'VeLoadingFlow',
  /* VE_LOADING_SWING: "VeLoadingSwing", */
  /*  VE_LOADING_CIRCLE: "VeLoadingCircle",
VE_LOADING_CIRCLE_FADE: "VeLoadingCircleFade", */
  VE_LOADING_GRID: 'VeLoadingGrid',
  VE_LOADING_FOLD: 'VeLoadingFold',
}

// spin name
export const SPIN_NAMES = {
  PLANE: 'plane',
  GRID: 'grid',
  WAVE: 'wave',
  FLOW: 'flow',
  BOUNCE: 'bounce',
  PULSE: 'pulse',
}
