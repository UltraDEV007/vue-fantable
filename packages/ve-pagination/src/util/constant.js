// prefix
export const PREFIX_CLS = 've-pagination-'

// locale comp name
export const LOCALE_COMP_NAME = 'pagination'

// emit events
export const EMIT_EVENTS = {
  PAGE_NUMBER_CHANGE: 'on-page-number-change',
  PAGE_SIZE_CHANGE: 'on-page-size-change',
}

// comps name
export const COMPS_NAME = {
  VE_PAGINATION: 'VePagination',
}
