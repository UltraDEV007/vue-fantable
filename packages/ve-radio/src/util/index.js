import { PREFIX_CLS } from './constant'

/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return PREFIX_CLS + cls
}
