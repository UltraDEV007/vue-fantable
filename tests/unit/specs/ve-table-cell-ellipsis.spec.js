import { mount } from "@vue/test-utils";
import veTable from "@/ve-table";
import { later } from "../util";

describe("veTable cell ellipsis", () => {
    const TABLE_DATA = [
        {
            name: "John",
            date: "1900-05-20",
            hobby: "coding and coding repeat",
            address:
                "No.1 Century Avenue, Shanghai,this is a long text,this is a long text,this is a long text,this is a long text,this is a long text,this is a long text",
            rowkey: 0,
        },
        {
            name: "Dickerson",
            date: "1910-06-20",
            hobby: "coding and coding repeat",
            address:
                "No.1 Century Avenue, Beijing,this is a long text,this is a long text,this is a long text,this is a long text",
            rowkey: 1,
        },
        {
            name: "Larsen",
            date: "2000-07-20",
            hobby: "coding and coding repeat",
            address:
                "No.1 Century Avenue, Chongqing,this is a long text,this is a long text,this is a long text,this is a long text,this is a long text,this is a long text,this is a long text,this is a long text",
            rowkey: 2,
        },
        {
            name: "Geneva",
            date: "2010-08-20",
            hobby: "coding and coding repeat",
            address:
                "No.1 Century Avenue, Xiamen,this is a long text,this is a long text",
            rowkey: 3,
        },
        {
            name: "Jami",
            date: "2020-09-20",
            hobby: "coding and coding repeat",
            address: "No.1 Century Avenue, Shenzhen",
            rowkey: 4,
        },
    ];

    const COLUMNS = [
        {
            field: "name",
            key: "a",
            title: "Name",
            align: "left",
            width: "15%",
        },
        {
            field: "date",
            key: "b",
            title: "Date",
            align: "left",
            width: "15%",
        },
        {
            field: "hobby",
            key: "c",
            title: "Hobby",
            align: "center",
            width: "30%",
        },
        {
            field: "address",
            key: "d",
            title: "Address",
            align: "left",
            width: "40%",
            ellipsis: {},
        },
    ];

    it("render single line ellipsis", () => {
        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                rowKeyFieldName: "rowKey",
            },
        });
        expect(wrapper.html()).toMatchSnapshot();
    });

    it("render multiline ellipsis", async () => {
        const wrapper = mount(veTable, {
            propsData: {
                columns: [
                    {
                        field: "name",
                        key: "a",
                        title: "Name",
                        align: "left",
                        width: "15%",
                    },
                    {
                        field: "date",
                        key: "b",
                        title: "Date",
                        align: "left",
                        width: "15%",
                    },
                    {
                        field: "hobby",
                        key: "c",
                        title: "Hobby",
                        align: "center",
                        width: "30%",
                    },
                    {
                        field: "address",
                        key: "d",
                        title: "Address",
                        align: "left",
                        width: "40%",
                        ellipsis: {
                            lineClamp: 3,
                        },
                    },
                ],
                // table data
                tableData: TABLE_DATA,
                rowKeyFieldName: "rowKey",
                fixedHeader: true,
            },
        });

        await later();

        const trEl = wrapper.findAll(".ve-table-body-tr").at(0);
        expect(
            trEl
                .findAll(".ve-table-body-td")
                .at(3)
                .find(".ve-table-body-td-span-ellipsis")
                .exists(),
        ).toBe(true);

        // doesn't support -webkit-line-clamp style attr
        /*    expect(
            trEl
                .findAll(".ve-table-body-td")
                .at(3)
                .find(".ve-table-body-td-span-ellipsis")
                .attributes("style")
        ).toBe("-webkit-line-clamp: 3;"); */
    });
});
