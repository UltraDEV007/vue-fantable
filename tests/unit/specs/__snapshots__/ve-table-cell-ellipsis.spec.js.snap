// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`veTable cell ellipsis > render single line ellipsis 1`] = `
"<div class="vue-table-root">
  <div style="width: 100%;" class="fan-table fan-table-border-around">
    <div class="fan-table-container fan-table-enable-cell-selection" style="height: 0px;">
      <!---->
      <div class="fan-table-content-wrapper">
        <table class="fan-table-content fan-table-border-x">
          <colgroup>
            <col style="width: 15%;">
            <col style="width: 15%;">
            <col style="width: 30%;">
            <col style="width: 40%;">
          </colgroup>
          <thead class="fan-table-fixed-header fan-table-header" tableviewportwidth="0">
            <tr class="fan-table-header-tr">
              <th style="text-align: left; top: 0px;" class="fan-table-header-th" rowspan="1" colspan="1">
                <!---->Name
                <!---->
                <!---->
                <!---->
              </th>
              <th style="text-align: left; top: 0px;" class="fan-table-header-th" rowspan="1" colspan="1">
                <!---->Date
                <!---->
                <!---->
                <!---->
              </th>
              <th style="text-align: center; top: 0px;" class="fan-table-header-th" rowspan="1" colspan="1">
                <!---->Hobby
                <!---->
                <!---->
                <!---->
              </th>
              <th style="text-align: left; top: 0px;" class="fan-table-header-th fan-table-last-column" rowspan="1" colspan="1">
                <!---->Address
                <!---->
                <!---->
                <!---->
              </th>
            </tr>
          </thead>
          <tbody class="fan-table-body fan-table-stripe fan-table-row-highlight">
            <tr style="height: 0px;">
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
            </tr>
            <tr class="fan-table-body-tr">
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="a">
                <!---->
                <!---->
                <!---->John
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="b">
                <!---->
                <!---->
                <!---->1900-05-20
              </td>
              <td class="fan-table-body-td" style="text-align: center;" rowspan="1" colspan="1" col-key="c">
                <!---->
                <!---->
                <!---->coding and coding repeat
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="d">
                <!---->
                <!---->
                <!----><span title="No.1 Century Avenue, Shanghai,this is a long text,this is a long text,this is a long text,this is a long text,this is a long text,this is a long text" style="-webkit-line-clamp: 1;" class="fan-table-body-td-span-ellipsis">No.1 Century Avenue, Shanghai,this is a long text,this is a long text,this is a long text,this is a long text,this is a long text,this is a long text</span>
              </td>
            </tr>
            <!---->
            <tr class="fan-table-body-tr">
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="a">
                <!---->
                <!---->
                <!---->Dickerson
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="b">
                <!---->
                <!---->
                <!---->1910-06-20
              </td>
              <td class="fan-table-body-td" style="text-align: center;" rowspan="1" colspan="1" col-key="c">
                <!---->
                <!---->
                <!---->coding and coding repeat
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="d">
                <!---->
                <!---->
                <!----><span title="No.1 Century Avenue, Beijing,this is a long text,this is a long text,this is a long text,this is a long text" style="-webkit-line-clamp: 1;" class="fan-table-body-td-span-ellipsis">No.1 Century Avenue, Beijing,this is a long text,this is a long text,this is a long text,this is a long text</span>
              </td>
            </tr>
            <!---->
            <tr class="fan-table-body-tr">
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="a">
                <!---->
                <!---->
                <!---->Larsen
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="b">
                <!---->
                <!---->
                <!---->2000-07-20
              </td>
              <td class="fan-table-body-td" style="text-align: center;" rowspan="1" colspan="1" col-key="c">
                <!---->
                <!---->
                <!---->coding and coding repeat
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="d">
                <!---->
                <!---->
                <!----><span title="No.1 Century Avenue, Chongqing,this is a long text,this is a long text,this is a long text,this is a long text,this is a long text,this is a long text,this is a long text,this is a long text" style="-webkit-line-clamp: 1;" class="fan-table-body-td-span-ellipsis">No.1 Century Avenue, Chongqing,this is a long text,this is a long text,this is a long text,this is a long text,this is a long text,this is a long text,this is a long text,this is a long text</span>
              </td>
            </tr>
            <!---->
            <tr class="fan-table-body-tr">
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="a">
                <!---->
                <!---->
                <!---->Geneva
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="b">
                <!---->
                <!---->
                <!---->2010-08-20
              </td>
              <td class="fan-table-body-td" style="text-align: center;" rowspan="1" colspan="1" col-key="c">
                <!---->
                <!---->
                <!---->coding and coding repeat
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="d">
                <!---->
                <!---->
                <!----><span title="No.1 Century Avenue, Xiamen,this is a long text,this is a long text" style="-webkit-line-clamp: 1;" class="fan-table-body-td-span-ellipsis">No.1 Century Avenue, Xiamen,this is a long text,this is a long text</span>
              </td>
            </tr>
            <!---->
            <tr class="fan-table-body-tr">
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="a">
                <!---->
                <!---->
                <!---->Jami
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="b">
                <!---->
                <!---->
                <!---->2020-09-20
              </td>
              <td class="fan-table-body-td" style="text-align: center;" rowspan="1" colspan="1" col-key="c">
                <!---->
                <!---->
                <!---->coding and coding repeat
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="d">
                <!---->
                <!---->
                <!----><span title="No.1 Century Avenue, Shenzhen" style="-webkit-line-clamp: 1;" class="fan-table-body-td-span-ellipsis">No.1 Century Avenue, Shenzhen</span>
              </td>
            </tr>
            <!---->
          </tbody>
          <tfoot class="fan-table-fixed-footer fan-table-footer"></tfoot>
        </table>
        <div class="fan-table-selection-wrapper">
          <div class="fan-table-selection-fixed-left">
            <!---->
            <!---->
            <!---->
            <!---->
          </div>
          <div class="fan-table-selection-middle">
            <!---->
            <!---->
            <!---->
            <!---->
          </div>
          <div class="fan-table-selection-fixed-right">
            <!---->
            <!---->
            <!---->
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div style="top: 0px; left: 0px; height: 1px; z-index: -1; opacity: 0;" class="fan-table-edit-input-container" allrowkeys=",,,,"><textarea class="fan-table-edit-input" tabindex="-1"></textarea></div>
    <!---->
    <!---->
  </div>
</div>"
`;

exports[`veTable cell ellipsis render single line ellipsis 1`] = `
<div class="vue-table-root">
  <div class="fan-table fan-table-border-around" style="width: 100%;">
    <div class="fan-table-container fan-table-enable-cell-selection" style="height: 0px;">
      <div class="fan-table-content-wrapper">
        <table class="fan-table-content fan-table-border-x">
          <colgroup>
            <col style="width: 15%;">
            <col style="width: 15%;">
            <col style="width: 30%;">
            <col style="width: 40%;">
          </colgroup>
          <thead class="fan-table-fixed-header fan-table-header">
            <tr class="fan-table-header-tr">
              <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: left; top: 0px;">Name</th>
              <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: left; top: 0px;">Date</th>
              <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: center; top: 0px;">Hobby</th>
              <th rowspan="1" colspan="1" class="fan-table-header-th fan-table-last-column" style="text-align: left; top: 0px;">Address</th>
            </tr>
          </thead>
          <tbody class="fan-table-body fan-table-row-hover fan-table-row-highlight">
            <tr style="height: 0px;">
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
            </tr>
            <tr class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: left;">John</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">1900-05-20</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;"><span title="No.1 Century Avenue, Shanghai,this is a long text,this is a long text,this is a long text,this is a long text,this is a long text,this is a long text" class="fan-table-body-td-span-ellipsis">No.1 Century Avenue, Shanghai,this is a long text,this is a long text,this is a long text,this is a long text,this is a long text,this is a long text</span></td>
            </tr>
            <tr class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: left;">Dickerson</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">1910-06-20</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;"><span title="No.1 Century Avenue, Beijing,this is a long text,this is a long text,this is a long text,this is a long text" class="fan-table-body-td-span-ellipsis">No.1 Century Avenue, Beijing,this is a long text,this is a long text,this is a long text,this is a long text</span></td>
            </tr>
            <tr class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: left;">Larsen</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">2000-07-20</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;"><span title="No.1 Century Avenue, Chongqing,this is a long text,this is a long text,this is a long text,this is a long text,this is a long text,this is a long text,this is a long text,this is a long text" class="fan-table-body-td-span-ellipsis">No.1 Century Avenue, Chongqing,this is a long text,this is a long text,this is a long text,this is a long text,this is a long text,this is a long text,this is a long text,this is a long text</span></td>
            </tr>
            <tr class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: left;">Geneva</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">2010-08-20</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;"><span title="No.1 Century Avenue, Xiamen,this is a long text,this is a long text" class="fan-table-body-td-span-ellipsis">No.1 Century Avenue, Xiamen,this is a long text,this is a long text</span></td>
            </tr>
            <tr class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: left;">Jami</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">2020-09-20</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;"><span title="No.1 Century Avenue, Shenzhen" class="fan-table-body-td-span-ellipsis">No.1 Century Avenue, Shenzhen</span></td>
            </tr>
          </tbody>
          <tfoot class="fan-table-fixed-footer fan-table-footer"></tfoot>
        </table>
        <div class="fan-table-selection-wrapper">
          <div class="fan-table-selection-fixed-left"></div>
          <div class="fan-table-selection-middle"></div>
          <div class="fan-table-selection-fixed-right"></div>
        </div>
      </div>
    </div>
    <div class="fan-table-edit-input-container" style="top: 0px; left: 0px; height: 1px; z-index: -1; opacity: 0;"><textarea tabindex="-1" class="fan-table-edit-input"></textarea></div>
  </div>
</div>
`;
